### PR TITLE
feat(1830): add build cache dir config to launcher

### PR DESCRIPTION
**Pull request by parthasl**
## Context

Pass cache directories to the launcher to store build cache to disk

## Objective

This PR creates environment variables for the pipeline, job and event cache directories which will be used in build to store folders/files as part of cache configuration.

## References

[1830](https://github.com/screwdriver-cd/screwdriver/issues/1830)

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: launch.go [hunk 1]
```diff
@@ -615,6 +620,26 @@ func main() {
...
+			Name:   "cache-strategy",
+			Usage:  "Cache strategy",
+			Value:  "",
+		},
+		cli.StringFlag{
+			Name:   "pipeline-cache-dir",
+			Usage:  "Pipeline cache directory",
+			Value:  "",
+		},
+        cli.StringFlag{
```
Comment (jithine) [inline launch.go]: indentation of this block is off.

File: launch.go [hunk 2]
```diff
@@ -629,6 +654,10 @@ func main() {
 		buildID, err := strconv.Atoi(c.Args().Get(0))
 		buildTimeoutSeconds := c.Int("build-timeout") * 60
 		fetchFlag := c.Bool("only-fetch-token")
+        cacheStrategy := c.String("cache-strategy")
```
Comment (jithine) [inline launch.go]: indentation of this block is off.

File: launch.go (hunk 1 above)
Comment (parthasl) [reply launch.go]: fixed

File: launch.go (hunk 2 above)
Comment (parthasl) [reply launch.go]: fixed

File: launch.go [hunk 3]
```diff
@@ -216,7 +216,7 @@ func convertToArray(i interface{}) (array []int) {
 	}
 }
 
-func launch(api screwdriver.API, buildID int, rootDir, emitterPath, metaSpace, storeURL, uiURL, shellBin string, buildTimeout int, buildToken string) error {
+func launch(api screwdriver.API, buildID int, rootDir, emitterPath, metaSpace, storeURL, uiURL, shellBin string, buildTimeout int, buildToken string, cacheStrategy string, pipelineCacheDir string, jobCacheDir string, eventCacheDir string) error {
```
Comment (DekusDenial) [inline launch.go]: you can do the short form `cacheStrategy, pipelineCacheDir, jobCacheDir, eventCacheDir string` for these new args are of type `string`

File: launch.go [hunk 4]
```diff
@@ -429,6 +429,10 @@ func launch(api screwdriver.API, buildID int, rootDir, emitterPath, metaSpace, s
 		"SD_STORE_URL":           fmt.Sprintf("%s/%s/", storeURL, "v1"),
 		"SD_UI_URL":              fmt.Sprintf("%s/", uiURL),
 		"SD_TOKEN":               buildToken,
+		"SD_CACHE_STRATEGY":      cacheStrategy,
```
Comment (DekusDenial) [inline launch.go]: so wanna make sure, so the default value for the strategy is set at the executor level?

File: launch.go [hunk 5]
```diff
@@ -502,10 +506,11 @@ func createEnvironment(base map[string]string, secrets screwdriver.Secrets, buil
 }
 
 // Executes the command based on arguments from the CLI
-func launchAction(api screwdriver.API, buildID int, rootDir, emitterPath, metaSpace, storeURI, uiURI, shellBin string, buildTimeout int, buildToken string) error {
+func launchAction(api screwdriver.API, buildID int, rootDir, emitterPath, metaSpace, storeURI, uiURI, shellBin string, buildTimeout int, buildToken string, cacheStrategy string, pipelineCacheDir string, jobCacheDir string, eventCacheDir string) error {
```
Comment (DekusDenial) [inline launch.go]: same as above

File: launch.go (hunk 4 above)
Comment (parthasl) [reply launch.go]: yes

File: launch.go (hunk 3 above)
Comment (parthasl) [reply launch.go]: done

File: launch.go (hunk 5 above)
Comment (parthasl) [reply launch.go]: done

**Pull request merged**